prow/pod-utils/gcs/upload.go

```diff
@@ -35,29 +35,44 @@ type UploadFunc func(obj *storage.ObjectHandle) error
 // Upload uploads all of the data in the
 // uploadTargets map to GCS in parallel. The map is
 // keyed on GCS path under the bucket
+// 3 attempts will be made to have an error free upload
 func Upload(bucket *storage.BucketHandle, uploadTargets map[string]UploadFunc) error {
+
 	errCh := make(chan error, len(uploadTargets))
 	group := &sync.WaitGroup{}
 	group.Add(len(uploadTargets))
+
 	for dest, upload := range uploadTargets {
+
 		obj := bucket.Object(dest)
 		logrus.WithField("dest", dest).Info("Queued for upload")
 		go func(f UploadFunc, obj *storage.ObjectHandle, name string) {
 			defer group.Done()
-			if err := f(obj); err != nil {
-				errCh <- err
+			retries := 3
```

Comment:
I would suggest making this a `const uploadRetries = 3` and using this same value in both this for loop as well as in the test case.

That way the test will continue working correctly if we decide that we actually want 2 or 5 retries.